### PR TITLE
Added RXRetrofitDemo

### PR DESCRIPTION
**Pull request by raghavsatyadev**
Added this demo which uses generalised model for calling API with Retrofit 2.0 & RxJava 2.0

### CODE REVIEW

**Comment by PareshMayani** [comment]
Thank you @raghavsatyadev and @WilsonChristian94 for putting efforts together, will perform review on it and share!

File: android/examples_demos/RXRetrofitDemo/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+
+def androidVersion = "25.3.1";
+def retrofitVersion = "2.3.0";
+dependencies {
+    compile fileTree(dir: 'libs', include: ['*.jar'])
+    compile "com.android.support:appcompat-v7:$androidVersion"
```
Comment (PareshMayani) [inline android/examples_demos/RXRetrofitDemo/app/build.gradle]: this variable name should be "supportLibVersion" actually

File: android/examples_demos/RXRetrofitDemo/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+
+def androidVersion = "25.3.1";
+def retrofitVersion = "2.3.0";
+dependencies {
+    compile fileTree(dir: 'libs', include: ['*.jar'])
```
Comment (PareshMayani) [inline android/examples_demos/RXRetrofitDemo/app/build.gradle]: FYI compile is getting deprecated from gradle 3.4.0. Just FYI :)

Comment (raghavsatyadev) [inline android/examples_demos/RXRetrofitDemo/app/build.gradle]: Thank you for the information.

File: android/examples_demos/RXRetrofitDemo/app/build.gradle (hunk 1 above)
Comment (raghavsatyadev) [inline android/examples_demos/RXRetrofitDemo/app/build.gradle]: Will update in next commit.

Comment (PareshMayani) [inline android/examples_demos/RXRetrofitDemo/app/build.gradle]: @raghavsatyadev waiting for your commit on it. Thanks

Comment (raghavsatyadev) [inline android/examples_demos/RXRetrofitDemo/app/build.gradle]: @PareshMayani Done sir.

**Comment by PareshMayani** [comment]
Thanks @raghavsatyadev for the contribution, looking forward to see more contribution!

**Pull request merged**